orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/cluster/DetermineRollbackCandidatesTask.java

```diff
@@ -281,6 +234,172 @@ public TaskResult execute(@Nonnull StageExecution stage) {
         .build();
   }
 
+  @Nullable
+  private List<ServerGroup> getServerGroups(
+      AtomicReference<Moniker> moniker, String credentials, String cloudProvider) {
+    return Optional.ofNullable(fetchClusterInfoWithRetry(moniker, credentials, cloudProvider))
+        .map(clusterInfo -> clusterInfo.get("serverGroups"))
+        .map(this::toServerGroups)
+        .orElse(null);
+  }
+
+  /** Deserialize a list of server groups. The list is sorted by creation time, newest first */
+  @Nonnull
+  private List<ServerGroup> toServerGroups(Object obj) {
+    List<ServerGroup> serverGroups =
+        objectMapper.convertValue(obj, new TypeReference<List<ServerGroup>>() {});
```

Comment:
nit: If this is a hot path, it may be worthwhile moving the `TypeReference` object into a static member of the class.

Reply:
Fixed in 512516fb42c4800ff6f0f9330f72510c097b6247